AutoTenez.py

```diff
@@ -171,170 +51,339 @@ def make_reservation(headers, cookies, date_tomorrow, md5slotkey, your_external_
         "Connection": "close",
         "Content-Type": "application/json",
     }
-    r = requests.get("https://api.socie.nl/public/pling", headers=headers)
-    cookies = r.cookies
-
-except ValueError as e:
-    print("ERROR! Failed to retrieve the cookies. Could not parse the response")
-    print(e)
-    sys.exit(-1)
-
-except Exception as e:
-    print("ERROR! An unexpected error occurred while trying to retrieve the cookies")
-    print(e)
-    sys.exit(-1)
-
-# Logging you in
-try:
-    print("Logging in as " + email_address + "...")
-    payload = {
-        "appType": "TENNIS",
-        "email": email_address,
-        "password": password,
-    }
 
-    r = requests.post("https://api.socie.nl/login/socie", json=payload, headers=headers, cookies=cookies)
-    response = r.json()
-    # Obtain bearer token from the JSON reponse
-    bearer_token = response['access_token']
-    # Add bearer token to all future requests
-    headers['Authorization'] = "bearer " + bearer_token
-
-# Catch JSONDecodeError
-except ValueError as e: 
-    print("ERRROR! Failed to parse the response")
-    print(e)
-    sys.exit(-1)
-
-except Exception as e:
-    print("ERROR! An unexpected error occurred while logging in")
-    print(e)
-    sys.exit(-1)
-
-try:
-    print("Retrieving necessary IDs...")
-    # Retrieve membership ID, which is included in the bearer token, so we can make another call to retrieve your external reference
-    decoded = jwt.decode(bearer_token, options={"verify_signature": False})
-    # Loop through all roles. Which is usually just one role
-    for role in decoded['roles']:
-        role_hash = role
+    def __init__(self, reservation_date, player2_external_reference, player3_external_reference, player4_external_reference):
+        print("Initializing AutoTenez at " + str(datetime.now()))
+        # Override class' reservation date if specified
```

Comment:
```suggestion
        # Set class' reservation date if specified. Otherwise, default to tomorrow
```
Furthermore, I think it is desired to print the reservation date if the else-statement is reached. Don't you agree?

Reply:
- Suggestion applied
- Added print statement in case of else